selfdrive/car/mazda/carstate.py

```diff
@@ -0,0 +1,235 @@
+    dt = 0.01
+
+    self.v_ego_kf = KF1D(x0=[[0.], [0.]],
+                         A=[[1., dt], [0.0, 1.]],
+                         C=[1., 0.],
+                         K=[[0.12287673], [0.29666309]])
+    
+    self.v_ego = 0.
+
+  def update(self, pt_cp, cam_cp):
+
+    self.can_valid = pt_cp.can_valid
+    #self.can_valid = True
+
+    self.v_wheel_fl = pt_cp.vl["WHEEL_SPEEDS"]['FL'] * CV.KPH_TO_MS
+    self.v_wheel_fr = pt_cp.vl["WHEEL_SPEEDS"]['FR'] * CV.KPH_TO_MS
+    self.v_wheel_rl = pt_cp.vl["WHEEL_SPEEDS"]['RL'] * CV.KPH_TO_MS
+    self.v_wheel_rr = pt_cp.vl["WHEEL_SPEEDS"]['RR'] * CV.KPH_TO_MS
+    v_wheel = (self.v_wheel_fl + self.v_wheel_fr + self.v_wheel_rl + self.v_wheel_rr) / 4.
+
+    self.v_ego_raw = v_wheel
+    # FIXME
+    v_ego_x = self.v_ego_kf.update(v_wheel)
+    self.v_ego = float(v_ego_x[0])
+    self.a_ego = float(v_ego_x[1])
+
+    self.prev_left_blinker_on = self.left_blinker_on
+    self.prev_right_blinker_on = self.right_blinker_on
+    self.prev_blinker_on = self.blinker_on
+    self.left_blinker_on = pt_cp.vl["BLINK_INFO"]['LEFT_BLINK'] == 1
+    self.right_blinker_on = pt_cp.vl["BLINK_INFO"]['RIGHT_BLINK'] == 1
+    self.blinker_on = self.left_blinker_on or self.right_blinker_on
+
+    self.acc_active = pt_cp.vl["CRZ_CTRL"]['CRZ_ACTIVE']
+    self.main_on = pt_cp.vl["CRZ_CTRL"]['CRZ_ACTIVE']
+
+    if self.acc_active != self.acc_active_last:
+      self.v_cruise_pcm =  v_wheel // CV.KPH_TO_MS
+      self.acc_active_last = self.acc_active
+
+    self.steer_torque_driver = pt_cp.vl["STEER_TORQUE"]['STEER_TORQUE_SENSOR']
+    self.steer_torque_motor = pt_cp.vl["STEER_TORQUE"]['STEER_TORQUE_MOTOR']
+    self.steer_override = abs(self.steer_torque_driver) > 15 #fixme
+
+    self.angle_steers = pt_cp.vl["STEER"]['STEER_ANGLE']
+    self.angle_steers_rate = pt_cp.vl["STEER_RATE"]['STEER_ANGLE_RATE']
+
+    #self.standstill = pt_cp.vl["PEDALS"]['STANDSTILL'] == 1
+    #self.brake_pressed = pt_cp.vl["PEDALS"]['BREAK_PEDAL_1'] == 1
+
+    self.standstill = self.v_ego_raw < 0.001
+
+    self.door_open = any([pt_cp.vl["DOORS"]['FL'],
+                          pt_cp.vl["DOORS"]['FR'],
+                          pt_cp.vl["DOORS"]['BL'],
+                          pt_cp.vl["DOORS"]['BR']])
+
+    self.seatbelt_unlatched =  pt_cp.vl["SEATBELT"]['DRIVER_SEATBELT'] == 0
+
+    self.steer_error = False
+    self.brake_error = False
+
+    self.steer_lkas.block = pt_cp.vl["STEER_RATE"]['LKAS_BLOCK']
+    self.steer_lkas.track = pt_cp.vl["STEER_RATE"]['LKAS_TRACK_STATE']
+    self.steer_lkas.handsoff = pt_cp.vl["STEER_RATE"]['HANDS_OFF_5_SECONDS']
+
+    #self.steer_not_allowed = self.steer_lkas.block == 1
+    self.low_speed_lockout = (v_wheel // CV.KPH_TO_MS) < 20
+
+    #if self.CAM_LT.ctr != cam_cp.vl["CAM_LANETRACK"]['CTR'] and cam_cp.vl["CAM_LANETRACK"]['CTR'] == cam_cp.vl["CAM_LKAS"]['CTR']:
+
+    self.cam_laneinfo = cam_cp.vl["CAM_LANEINFO"]
+
+    if self.CAM_LKAS.ctr != cam_cp.vl["CAM_LKAS"]['CTR']:
```

Comment:
why you can't simply do:
```
self.CAM_LKAS = cam_cp.vl["CAM_LKAS"]
```
regardless the counter value?

Reply:
Again, in early development I couldn't get the car to steer, so I had code at some to sync with the stock LKAS ctr. Will drop.